### PR TITLE
adjust resources to CamelCase

### PR DESCRIPTION
**Pull request by NHingerl**
**Description**

Changes proposed in this pull request:

- Change the spelling of K8s and custom resources to CamelCase

**Related issue(s)**

https://github.com/kyma-project/community/pull/693
https://github.com/kyma-project/community/issues/680

Mostly reverts https://github.com/kyma-project/kyma/pull/13668, https://github.com/kyma-project/kyma/pull/13647, https://github.com/kyma-project/kyma/pull/13634.

### CODE REVIEW

**Comment by NHingerl** [comment]
First commit took care of: 
Addons Configuration
API Rule
Application Mapping
Asset Group
Certificate Request
Cluster Asset
Cluster Asset Group
Cluster Bucket
Cluster Role
Cluster Role Binding
Cluster   Service Class
Compass   Connection
Config Map
Cron Job
Custom Resource Definition
Daemon Set
Destination Rule
Egress   Rule
Event   Subscription
Git Repository
Horizontal Pod Autoscaler
Istio Operator
Limit Ranges
Log Pipeline
Network Policy
Open Service Broker
Persistent Volume
Persistent Volume Claim
Pod Preset
Prometheus Rule
Replica Set
Role Binding
Service Account
Service Binding
Service Binding Usage
Service Instance
Service Monitor
Target Kind
Test Definition
Token Request
Usage Kind
Virtual Service
Volume Snapshot
Volume Snapshot Class

File: README.md [hunk 1]
```diff
@@ -109,7 +109,7 @@ If you would like to join us and work together on the Kyma project, there are so
   - Downloading additional tools modules with the `go get` command
 
 - Kubernetes basic skillset:
-  - Understanding basic [Kubernetes architecture](https://shipit.dev/posts/kubernetes-overview-diagrams.html) and basic concepts such as: Namespace, Pod, Deployment, Secret, Config Map, Replica Set, Service, Custom Resource Definition, Kubernetes Control Loop; understanding Kubernetes Design Patterns such as sidecars and init containers
+  - Understanding basic [Kubernetes architecture](https://shipit.dev/posts/kubernetes-overview-diagrams.html) and basic concepts such as: Namespace, Pod, Deployment, Secret, ConfigMap, ReplicaSet, Service, CustomResourc Definition, Kubernetes Control Loop; understanding Kubernetes Design Patterns such as sidecars and init containers
```
Comment (medmes) [inline README.md:112]: `CustomResource`

Comment (majakurcius) [inline README.md:112]:
```suggestion
  - Understanding basic [Kubernetes architecture](https://shipit.dev/posts/kubernetes-overview-diagrams.html) and basic concepts such as: Namespace, Pod, Deployment, Secret, ConfigMap, ReplicaSet, Service, CustomResourceDefinition, Kubernetes Control Loop; understanding Kubernetes Design Patterns such as sidecars and init containers
```

Comment (NHingerl) [reply README.md]: Thank you for spotting this!

**Comment by NHingerl** [comment]
2nd commit checked for the context of the words "CR", "custom resource", and for the following additional terms:

- Log Parser
- DNS Provider
- DNS Entry
- Eventing Backend

File: components/README.md [hunk 2]
```diff
@@ -15,13 +15,13 @@ This table lists the available types:
...
 | **job** | A [Kubernetes Job](https://kubernetes.io/docs/tasks/job/) which performs a task once or periodically. It uses the name of the task it performs. |istio-patch-job (not renamed yet)|
 | **proxy** | Acts as a proxy for an existing component, usually introducing a security model for this component. It uses the component's name. | - |
 | **service** | Serves an HTTP/S-based API, usually securely exposed to the public. It uses the domain name and the API it serves.| - |
-| **broker** | Implements the [Open Service Broker](https://www.openservicebrokerapi.org/) specification to enrich the Kyma Service Catalog with the services of a provider. It uses the name of the provider it integrates with.| azure-broker |
+| **broker** | Implements the [OpenServiceBroker](https://www.openservicebrokerapi.org/) specification to enrich the Kyma Service Catalog with the services of a provider. It uses the name of the provider it integrates with.| azure-broker |
```
Comment (grego952) [inline components/README.md:24]: Are you sure OpenServiceBroker should be in camelcase?

File: components/binding/README.md [hunk 3]
```diff
@@ -20,13 +19,13 @@ helm install binding ./charts/binding --wait
 
 Go through the short tutorial that presents the useage and capabilities of the component.
 
-1. Register a Target Kind. It's a resource that defines how and where environments can be injected.
+1. Register a TargetKind. It's a resource that defines how and where environments can be injected.
```
Comment (grego952) [inline components/binding/README.md:22]: Are you sure TargetKind should be in camelcase?

File: docs/05-technical-reference/ra-01-configuring-runtime.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 title: Configuring the Runtime
 ---
 
-> **NOTE:** To represent API and Event Definitions of the Applications connected to a Runtime, we recommend that you use the Open Service Broker API.
+> **NOTE:** To represent API and Event Definitions of the Applications connected to a Runtime, we recommend that you use the OpenServiceBroker API.
```
Comment (grego952) [inline docs/05-technical-reference/ra-01-configuring-runtime.md:5]: Are you sure OpenServiceBroker should be in camelcase?

File: components/README.md (hunk 2 above)
Comment (NHingerl) [reply components/README.md:24]: I'm absolutely not :D
It was one of the changes in https://github.com/kyma-project/kyma/pull/13634/files which I'm basically reverting, but I might have misunderstood why it was changed. 

In their own docs, they use blank spaces. But isn't it a resource, which we'd put in camel case? This seems like two style guides clashing.

File: components/binding/README.md (hunk 3 above)
Comment (NHingerl) [reply components/binding/README.md:22]: I'm sure about nothing here :D
It's called "a resource". And it was one of the changes from https://github.com/kyma-project/kyma/pull/13634/files which I tried to revert.

File: docs/05-technical-reference/ra-01-configuring-runtime.md (hunk 4 above)
Comment (NHingerl) [reply docs/05-technical-reference/ra-01-configuring-runtime.md:5]: Nope :D But whatever we do, let's be consistent here and in https://github.com/kyma-project/kyma/pull/14991#discussion_r937691774

**Review by grego952 — APPROVED** [review]

File: docs/05-technical-reference/ra-01-configuring-runtime.md (hunk 4 above)
Comment (majakurcius) [reply docs/05-technical-reference/ra-01-configuring-runtime.md:5]: This doesn't look like something that should be CamelCased. I checked in the [old Kyma documentation](https://kyma-project-old.netlify.app/docs/components/runtime-agent/#details-configuring-the-runtime) as well as their [website](https://www.openservicebrokerapi.org/) (and Google), and it's written with spaces in all of them.

Comment (NHingerl) [reply docs/05-technical-reference/ra-01-configuring-runtime.md]: updated with https://github.com/kyma-project/kyma/pull/14991/commits/7cb968371707df499a7fdc4bb2bfb64852cc1740

**Review by majakurcius — APPROVED** [review]
LGTM :)

**Pull request merged**